frontend/src/pages/feegrant/ChainGrants.js

```diff
@@ -0,0 +1,239 @@
+  );
+  const chainInfo = useSelector(
+    (state) => state.wallet?.networks?.[chainID]?.network
+  );
+  const currency = chainInfo?.config?.currencies[0];
+  const [tab, setTab] = useState(0);
+  const [selected, setSelected] = React.useState({});
+  const [infoOpen, setInfoOpen] = React.useState(false);
+
+  const handleTabChange = (value) => {
+    setTab(value);
+  };
+  const handleInfoClose = (value) => {
+    setInfoOpen(false);
+  };
+
+  return chainGrantsByMe?.length || chainGrantsToMe?.length ? (
+    <Card elevation={0} sx={{ p: 1, mt: 2 }}>
+      <CardContent>
+        <Grid container>
+          <Grid item>
+            <div
+              style={{
+                display: "flex",
+              }}
+            >
+              <Avatar
+                src={chainInfo.logos.menu}
+                sx={{
+                  width: 36,
+                  height: 36,
+                }}
+              />
+              <Typography
+                align="left"
+                variant="h6"
+                gutterBottom
+                color="text.secondary"
+                sx={{
+                  ml: 1,
+                  "&:hover": {
+                    cursor: "pointer",
+                  },
+                }}
+                onClick={() => navigate(`/${chainName.toLowerCase()}/feegrant`)}
+              >
+                {chainName.charAt(0).toUpperCase() + chainName.slice(1)}
+              </Typography>
+            </div>
+          </Grid>
+        </Grid>
+      </CardContent>
+      {selected?.allowance ? (
+        <FeegrantInfo
+          authorization={selected}
+          displayDenom={currency?.coinDenom}
+          open={infoOpen}
+          onClose={handleInfoClose}
+          coinDecimals={currency?.coinDecimals}
+        />
+      ) : (
+        <></>
+      )}
+      <GroupTab
+        tabs={[
+          {
+            title: "Granted By Me",
+          },
+          {
+            title: "Granted To Me",
+          },
+        ]}
+        handleTabChange={handleTabChange}
+      />
+      <TabPanel value={tab} index={0} key={"by-me"}>
+        <>
+          {!chainGrantsByMe?.length ? (
+            <Typography
+              variant="h6"
+              color="text.primary"
+              style={{
+                display: "flex",
+                justifyContent: "center",
+                padding: 16,
+              }}
+            >
+              No Feegrant found
+            </Typography>
+          ) : (
+            <>
+              <Table
+                sx={{ minWidth: 650 }}
+                aria-label="simple table"
+                size="small"
+              >
+                <TableHead>
+                  <StyledTableRow>
+                    <StyledTableCell>Grantee</StyledTableCell>
+                    <StyledTableCell>Type</StyledTableCell>
+                    <StyledTableCell>Expiration</StyledTableCell>
+                    <StyledTableCell>Details</StyledTableCell>
```

Comment:
Lets add revoke feegrant action in `Grants by me` tab.

Reply:
done